servers/rendering/renderer_canvas_cull.cpp

```diff
@@ -2543,6 +2545,25 @@ void RendererCanvasCull::_update_dirty_item(Item *p_item) {
 	p_item->update_dependencies = false;
 }
 
+void RendererCanvasCull::Canvas::set_use_linear_color(bool p_use_linear_color) {
+	use_linear_color = p_use_linear_color;
+
+	for (ChildItem &child_item : child_items) {
+		_set_item_use_linear_color(child_item.item, use_linear_color);
```

Comment:
I'm not sure about the complexity added here. 

Could you explain a bit about why it is necessary? In the meeting I was thinking that we could get away with checking the current canvas/viewport when updating the instance uniform. If that isn't the case we may need to change the paradigm a bit (and do something like what the global uniforms _should_ do which is store the color in both color spaces)

Reply:
The highlighted code should almost never be called (unless the user likes to toggle hdr2d every frame), most of the time the parent's boolean is copied down when setting parent.

Each canvas can have multiple viewports, each viewport may or may not be hdr2d, so I think this is the simplest and most performant solution given the state of things. I can't imagine a scenario where the same canvas would be rendered both in linear and srgb viewports at the same time?